source/1_engine/rex_engine/include/rex_engine/data_type.h

```diff
@@ -2,68 +2,73 @@
 
 namespace rex
 {
-    class DataType
+  class DataType
+  {
+  public:
+    enum class Value
     {
-    public:
-        enum class Value
-        {
-            NONE = 0,
+      NONE = 0,
 
-            INT8,
-            INT16,
-            INT32,
-            INT64,
+      Int8,
+      Int16,
+      Int32,
+      Int64,
 
-            UNSIGNED_INT8,
-            UNSIGNED_INT16,
-            UNSIGNED_INT32,
-            UNSIGNED_INT64,
+      UnsignedInt8,
+      UnsignedInt16,
+      UnsignedInt32,
+      UnsignedInt64,
 
-            FLOAT32,
-            FLOAT64,
+      Float32,
+      Float64,
 
-            MAT3,
-            MAT4,
+      Mat3,
+      Mat4,
 
-            VEC2,
-            VEC3,
-            VEC4,
+      Vec2,
+      Vec3,
+      Vec4,
 
-            IVEC2,
-            IVEC3,
-            IVEC4,
+      Ivec2,
+      Ivec3,
+      Ivec4,
 
-            UIVEC2,
-            UIVEC3,
-            UIVEC4,
+      Uvec2,
+      Uvec3,
+      Uvec4,
 
-            BOOL,
+      Bool,
 
-            SAMPLER2D,
-            SAMPLER3D,
-            SAMPLERCUBE,
+      Sampler2D,
+      Sampler3D,
+      SamplerCube,
 
-            STRING
-        };
+      String
+    };
 
-    public:
-        DataType(Value value);
-        
-        DataType(const DataType& other) = default;
-        DataType(DataType&& other) noexcept = default;
-        
-        DataType& operator= (const DataType& other);
-        DataType& operator= (DataType&& other) noexcept;
+  public:
+    explicit DataType(Value value);
+    ~DataType() = default;
 
-        bool operator==(const DataType& other) const;
-        bool operator!=(const DataType& other) const;
+    DataType(const DataType& other)     = default;
+    DataType(DataType&& other) noexcept = default;
 
-        Value get() const;
+    DataType& operator=(const DataType& other);
+    DataType& operator=(const Value& other);
```

Comment:
not sure what the point of this is?

Reply:
Because the ctor is explicit, this makes it possible to assign the enum directly. It makes the usage of the class a lot cleaner and easier. Maybe I needed a NOLINT in the first place.